src/runtime-pmix.c

```diff
@@ -106,13 +106,43 @@ shmem_runtime_get_rank(void)
     return myproc.rank;
 }
 
-
 int
 shmem_runtime_get_size(void)
 {
     return size;
 }
 
+int
+shmem_runtime_get_local_rank(int pe)
+{
+#ifdef USE_ON_NODE_COMMS
+    pmix_status_t rc;
+    pmix_proc_t proc;
+    pmix_value_t *val;
+    uint16_t local_rank;
+
+    (void)strncpy(proc.nspace, myproc.nspace, PMIX_MAX_NSLEN);
+    proc.rank = myproc.rank;
+    if (PMIX_SUCCESS == (rc = PMIx_Get(&proc, PMIX_LOCAL_RANK, NULL, 0, &val))) {
+        local_rank = val->data.uint16;
+        PMIX_VALUE_RELEASE(val);
+    } else {
+        RETURN_ERROR_MSG_PREINIT("PMIX_LOCAL_RANK is not properly initiated (%d) on PE %d\n", rc, pe);
+    }
+    return local_rank;
+#elif defined(USE_MEMCPY)
+    return pe == myproc.rank ? 0 : -1;
+#else
+    return -1;
+#endif
+}
```

Comment:
Yes, it can be that way with `USE_MEMCPY` - or are you asking about the other cases?   Either way, I'm not sure I fully understand this problem yet.  Would changing `shmem_runtime_get_local_size` similar to this help?

```c
int shmem_runtime_get_local_size(void) {
#ifdef USE_ON_NODE_COMMS
    return local_size;
#elif defined(USE_MEMCPY)
    return 1;
#else
    return 0;
#endif
}
```

Reply:
To keep things consistent I added "node" rank/size as well as "local" rank/size.  The "node" values depend on on-node comms and `USE_MEMCPY`, whereas the latter does not.  Hopefully that fixes the problem without causing too much confusion...